javascript/ql/src/semmle/javascript/NodeModuleResolutionImpl.qll

```diff
@@ -96,8 +96,11 @@ File resolveMainModule(PackageJSON pkg, int priority) {
     )
   )
   or
-  result =
-    tryExtensions(pkg.getFile().getParentContainer(), "index", priority - prioritiesPerCandidate())
+  exists(Folder folder | folder = pkg.getFile().getParentContainer() |
+    result =
+      tryExtensions([folder, folder.getChildContainer("src")], "index",
```

Comment:
It is quite common to name the `src` folder `lib` in JavaScript projects. But perhaps that is not the case with `lib/index.js`?

Reply:
[`lib`](https://github.com/search?q=extension%3Ajs+path%3Alib%2Findex&type=Code&ref=advsearch&l=&l=) is less common than [`src`](https://github.com/search?q=extension%3Ajs+path%3Asrc%2Findex&type=Code&ref=advsearch&l=&l=).   
It seems reasonable to add it, so I'll do that.